doc/_src_docs/applications/Mixed_Hier_usage.rstx

```diff
@@ -0,0 +1,103 @@
+.. _Mixed Integer and Hierarchical Variables Types Specifications: 
+
+Mixed Integer and Hierarchical usage (Variables, Sampling and Context)
+======================================================================
+
+Mixed integer variables types
+-----------------------------
+
+SMT provides the ``mixed_integer`` module to adapt existing surrogates to deal with categorical (or enumerate) and ordered variables using continuous relaxation.
+For ordered variables, the values are rounded to the nearest values from a provided list. If, instead, bounds are provided, the list will consist of all integers between those bounds.
+
+The user specifies x feature types through a list of types to be either:
+
+- ``FLOAT``: a continuous feature,
+- ``ORD``: an ordered valued feature,
+- or a tuple ``(ENUM, n)`` where n is the number of levels of the catagorical feature (i.e. an enumerate with n values)
+
+In the case of mixed integer sampling, bounds of each x feature have to be adapted to take into account feature types. While ``FLOAT`` and ``ORD`` feature still have an interval [lower bound, upper bound], the ``ENUM`` features bounds is defined by giving the enumeration/list of possible values (levels). 
+
+For instance, if we have the following ``xtypes``: ``[FLOAT, ORD, (ENUM, 2), (ENUM, 3)]``, a compatible ``xlimits`` could be ``[[0., 4], [-10, 10], ["blue", "red"], ["short", "medium",  "long"]]``.
+
+However, the functioning of ``ORD`` is twofold. As previously mentioned, it can be used like [lower bound, upper bound], in this case [0,5] will corresponds to [0,1,2,3,4,5]. But, on the other hand, ``ORD`` can be used as an enumeration/list of possible values (levels), in this case ["0","5","6"] will corresponds to [0,5,6]. Details can be found in [1]_ .
+
+Hierarchical variables roles
+----------------------------
+
+The ``mixed_integer`` module uses the framework of Audet et al. [2]_ to manage both mixed variables and hierarchical variables. We distinguish dimensional (or meta) variables which are a special type of variables that may affect the dimension of the problem and decide if some other decreed variables are included or excluded. The variable size problem can also includes neutral variables that are always included and actives. 
+
+The user specifies x feature role through a list of roles amongst:
+
+- ``META``: a dimensional feature,
+- ``DECREED``: an ordered or continuous decreed feature: either included or excluded in the variable-size problem,
+- ``NEUTRAL``: a neutral feature, part of the fixed-size problem
+
+Note that we do not consider decreed categorical variable. The decreed variables are always continuous or ordered.
+
+Mixed and hierarchical specifications
+-------------------------------------
+
+The ``XSpecs`` class helps implements the types, limits and roles of each variables as follows.
+
+  .. autoclass:: smt.utils.kriging.XSpecs
```

Comment:
See generated doc. role is missing in description  and xroles parameter is not rendered properly.